### ISSUE TITLE
ScanMyTesla  "The SSL connection could not be established, see inner exception."

### ISSUE THREAD

**Issue by bassmaster187** [bassmaster187/TeslaLogger#699]
![image](https://user-images.githubusercontent.com/6816385/144189788-99139fd2-c610-4d83-a51f-ce6211053647.png)

Certificate has been renewed on November 28th, 2021 and it seems like some devices run in certificate errors.

Certificate seems to be valid:
https://www.sslshopper.com/ssl-checker.html#hostname=teslalogger.de

![image](https://user-images.githubusercontent.com/6816385/144193670-5291bf00-80a6-4fad-b902-d2f566cb7a2d.png)

This issue is to collect informations about the issue and how to solve it.

I have a Samsung S21+ with latest Android 12. In Scan My Tesla V1.9.7 I receive the error metion before. Although teslalogger.de has a valid certificate in Chrome browser.

**Comment by ostorezio** [comment]
Hi, I had the same bug on 3 devices.
Two Samsung mobiles: A20e and A03s
And one tablet, Galaxy-Tab-A
A03s has Android 11, the others, if you need I have to check.

**Comment by eivissacopter** [comment]
Same SSL connection issue here in four different cars all running SMT 1.9.7 on Samsung Galaxy Tab A7 Lite T225 / Android 11.

**Comment by ostorezio** [comment]
X-posted ... somewhere else :-)
How about implementing the possibility to save the logged data locally into the smartphone memory?
This solution would overcome outages like this one or, more commonly, situations where the car is out of data coverage.

**Comment by eivissacopter** [comment]
Latest SMT Android Beta works fine on the same device (Samsung Galaxy Tab A7 Lite T225)

**Comment by ostorezio** [comment]
Where/how do I get the Beta?

**Comment by eivissacopter** [comment]
Google play Store / Scan My Tesla / Scroll down -> Join Beta -> Wait 10 Minutes -> Update app on play store.

**Comment by ostorezio** [comment]
GOT IT! Nice graphics, very cool.

How do I know that the TCP-IP/SSL communication is on? Can't find the internet link icon of previous versions, you know the little cloud with dots or crosses.
In fact during the test I turned off the WiFi in my mobile, and nothing happened, no errors.

**Comment by eivissacopter** [comment]
Check Teslalogger Settings to see when the token was last received. Also the whole dashboard is filling with errors if the connection doesn't work.

**Comment by ostorezio** [comment]
Many thanks, I will check

**Comment by bassmaster187** [comment]
It seems like it is a known bug in xamarin. I made a workaround and the new release will be published as soon as possible.

https://github.com/xamarin/xamarin-android/issues/6351

**Comment by Adminius** [comment]
Not a Beta SMT latest version Moto G4 with Android 5 no problems

**Comment by uniquepurpose** [comment]
I am having the same issue on a OnePlus 6T running Android 11 (OxygenOS 11.1.1.1 with security patch 2021-09-02) with SMT 1.9.7 (non-beta)
Same on a Gigaset GS4 running Android 10 (with security patch 2021-08-01) with SMT 1.9.7 (non-beta).

A workaround (disable CA "Digital Signature Trust Co. - DST Root CA X3") is available here:
https://github.com/xamarin/xamarin-android/issues/6351#issuecomment-932051312

**Comment by ostorezio** [comment]
@eivissacopter 
>Check Teslalogger Settings to see when the token was last received. Also the whole dashboard is filling with errors if the connection doesn't work.

I did check at the following screenshot
[Screenshot_20211203-151519.pdf](https://github.com/bassmaster187/TeslaLogger/files/7650303/Screenshot_20211203-151519.pdf)

Can't see the date of last received

**Comment by ostorezio** [comment]
Yesterday I stopped for a supercharging session.
The data Teslalogger shows for the session is very much reduced in Beta 2.3.0 compared to 1.9.7
Still a server connection problem?

**Comment by KermsGit** [comment]
On my Oneplus phones worked this workaround:

```
On the Android you can manualy disable the certificate "Digital Signature Trust Co. - DST Root CA X3".

Go to "Settings > Security > Encryption & credentials" > Trusted credentials"
Scroll down and disable "Digital Signature Trust Co. - DST Root CA X3"_

```
The SSL connection worked again after disable this certificate.

**Comment by Proffesor94** [comment]
> On my Oneplus phones worked this workaround:
> 
> ```
> On the Android you can manualy disable the certificate "Digital Signature Trust Co. - DST Root CA X3".
> 
> Go to "Settings > Security > Encryption & credentials" > Trusted credentials"
> Scroll down and disable "Digital Signature Trust Co. - DST Root CA X3"_
> ```
> 
> The SSL connection worked again after disable this certificate.

Confirmed, it runs for my mi note 10. Thanks.

**Comment by ostorezio** [comment]
Thanks uniquepurpose 
I downgraded SMT from Beta 2.3.0 to 1.9.7 and applied your workaround.
SMT is back in business!!!

**Comment by uniquepurpose** [comment]
> I downgraded SMT from Beta 2.3.0 to 1.9.7 and applied your workaround. SMT is back in business!!!
I tried the workaround yesterday with beta 2.3.0 and it works for me as well.
Kinda like the new UI.

**Comment by ostorezio** [comment]
Hmmm .... 2.3.0 doesn't work for me, see above.
Is that because I didn't apply the workaround on 2.3.0?
How do you know from 2.3.0 if data link is active or not?

**Comment by bassmaster187** [comment]
Sorry for the delay. The bug was fixed within one day, but google has decline our update because we had to compile the app against the newest Android SDK and that brought new bugs. I seems like we can release it tonight.

**Comment by markab0572** [comment]
same problem here, glad to hear that you could fix it. Wait for the update now. 
will change to Android 12
Google Pixel 4 with Android 11 and SMT 1.9.7
![IMG_1452](https://user-images.githubusercontent.com/2920181/145566173-4897485d-ab1d-4dda-a35b-84b789b4d5bf.jpeg)

**Comment by bassmaster187** [comment]
Fixed in Scan My Tesla 1.9.8
